### ISSUE TITLE
About the code.

### ISSUE THREAD

**Issue by Kurumi233** [unique-chan/Complement-Cross-Entropy#2]
Compare with the original COT code, there is no difference except the hyper-parameter gamma. But the balancing factor 1/(k-1) is not shown. Can you explain it? And I wonder how the results of the training process in COT and CCE are different? Thanks.

**Comment by unique-chan** [comment]
COT is proposed by Chen et al (2019). Their code is available at https://github.com/henry8527/COT/blob/master/code/COT.py. In fact, "loss /= float(self.classes)" is equal to "loss /= balancing factor" (or loss *= (1 / balancing factor)). In my opinion, "loss /= float(self.classes)" should be modified like ""loss /= (float(self.classes) - 1)", as indicated in their paper. However, COT is not my code, so I decided to not modify their logic in Python.

The key difference between COT and CCE is the number of parameter updates. COT requires twice updates per each iteration; meanwhile, CCE needs one update per each iteration. In every iteration, to be precise, COT once updates parameters with cross-entropy loss and then updates again with complement-entropy, which incurs training time almost doubled. 

CCE combines (+) cross-entropy with (-) complement-entropy. CCE also works well as COT without requiring additional training time.
